dot/parachain/prospective-parachains/prospective-parachains.go

```diff
@@ -4,14 +4,140 @@ import (
 	"context"
 	"errors"
 
+	"github.com/ChainSafe/gossamer/dot/parachain/backing"
+	fragmentchain "github.com/ChainSafe/gossamer/dot/parachain/prospective-parachains/fragment-chain"
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
 	"github.com/ChainSafe/gossamer/internal/log"
+	common "github.com/ChainSafe/gossamer/lib/common"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "prospective_parachains"), log.SetLevel(log.Debug))
 
+type RelayBlockViewData struct {
+	// The fragment chains for current and upcoming scheduled paras.
+	FragmentChains map[parachaintypes.ParaID]fragmentchain.FragmentChain
+}
+
+type View struct {
+	// Per relay parent fragment chains. These includes all relay parents under the implicit view.
+	PerRelayParent map[common.Hash]RelayBlockViewData
+	// The hashes of the currently active leaves. This is a subset of the keys in
+	// `per_relay_parent`.
+	ActiveLeaves map[common.Hash]struct{}
+	// The backing implicit view.
+	ImplicitView backing.ImplicitView
+}
+
+// Initialize with empty values.
+func NewView() *View {
+	return &View{
+		PerRelayParent: make(map[common.Hash]RelayBlockViewData),
+		ActiveLeaves:   make(map[common.Hash]struct{}),
+		ImplicitView:   nil, // TODO: currently there's no implementation for ImplicitView, reference is: https://github.com/paritytech/polkadot-sdk/blob/028e61be43f05f6f6c88c5cca94160f8db075585/polkadot/node/subsystem-util/src/backing_implicit_view.rs#L40
+	}
+}
+
+// Get the fragment chains of this leaf.
+func (v *View) GetFragmentChains(leaf common.Hash) map[parachaintypes.ParaID]fragmentchain.FragmentChain {
+	if viewData, ok := v.PerRelayParent[leaf]; ok {
+		return viewData.FragmentChains
+	}
+	return nil
+}
+
+func HandleIntroduceSecondedCandidate(
```

Comment:
I would suggest to make it a private method of `ProspectiveParachains` and remove the `handle` prefix like:

```go
func (pp * ProspectiveParachains) introduceSecondedCandidate(
```

Reply:
done!